### ISSUE TITLE
Setting an undefined value does not trigger a re-render.

### ISSUE THREAD

**Issue by kesteinbakk** [CharlesStover/reactn#51]
Maybe it's just me being stupid, but I had a hard time figuring out why my components did not update (with no errors or warnings) when I changed the global state. Turne out I initially had set the global state like this:
```
setGlobal({
  status: undefined,
  data: {},
})
```

Changes to the "status" item did not trigger any component updates. Solved by changing the initial state to

```
setGlobal({
  status: false, // null also works
  data: {},
})
```

A small comment about this in the docs? Or maybe it's totally obvious to everyone else :-)

**Comment by CharlesStover** [comment]
I think the current behavior is that `undefined` removes the property from the state instead of just setting the value to undefined. This is likely the source of the miscommunication between state and Components.

I am working on React 2.0 right now, which does _not_ treat `undefined` as a special case. This should no longer be an issue. I'll try to remember to include unit tests specifically for this case to verify.

Thanks for bringing it to my attention.

**Comment by CharlesStover** [comment]
Closing this as a potential duplicate of #57.

If the issue is as described in my previous comment, this is fixed in 2.0.
If the issue is as described in #57, it is `wontfix` for the reasons described there.

Thank you for the feedback and involvement in the project!
